### PR TITLE
Fix/embed routing and404

### PR DESCRIPTION
**Pull request by geckoslair**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -2,9 +2,8 @@
 <html lang="en">
   <head>
     <meta charset="utf-8">
-    <title>Paratii</title>
+    {{meta}}
```
Comment (bent0b0x) [inline src/index.html]: If we are calling this `meta` then should every `<meta>` tag be moved into a variable? We still have `<meta name="viewport">` below

File: src/server/app.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 const express = require('express')
-const path = require('path')
+// const path = require('path')
```
Comment (bent0b0x) [inline src/server/app.js]: Can you remove this comment?

File: src/server/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      if (video !== null) {
+        meta = openGraphHead(meta, video)
+        meta = twitterCardHead(meta, video)
+      } else {
+        meta = notFoundVideo(meta)
+      }
+
+      break
+    default:
+      console.log('default')
```
Comment (bent0b0x) [inline src/server/routes/index.js]: Mind deleting this `console.log` (and any others)?

File: src/server/routes/index.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      meta = basicHead(meta)
+      script = '<script type="text/javascript" src="/bundle.js"></script>'
+      break
+  }
+
+  // console.log(index)
```
Comment (bent0b0x) [inline src/server/routes/index.js]: Remove comment

File: src/server/routes/index.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  index = index.replace('{{meta}}', meta)
```
Comment (bent0b0x) [inline src/server/routes/index.js]: Is this just a string replacement? I would rather us not do this. Can't we use [express-handlebars](https://github.com/ericf/express-handlebars) instead of fudging together our own template engine?

File: src/server/routes/index.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  index = index.replace('{{script}}', script)
+  console.log(meta)
+  console.log(script)
```
Comment (bent0b0x) [inline src/server/routes/index.js]: Remove logging

File: src/server/routes/index.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  meta += `<meta property="og:video:secure_url" content="${embedUrl}">`
+  meta += `<meta property="og:video:type" content="text/html">`
+  meta += `<meta property="og:video:width" content="${width}" />`
+  meta += `<meta property="og:video:height" content="${height}" />`
+  meta += `<meta property="og:type" content="video.other" />`
+  meta += `<meta property="og:url" content="${embedUrl}" />`
+  meta += `<meta property="og:title" content="${video.title}" />`
+  meta += `<meta property="og:image" content="${thumbnailUrl}" />`
+  meta += `<meta property="og:description" content="${video.description}" />`
+
```
Comment (bent0b0x) [inline src/server/routes/index.js]: I think it would be best if we use a template engine and move all of this into a template partial, rather than a large string concatenation

File: src/server/app.js [hunk 8]
```diff
@@ -28,12 +28,15 @@ if (process.env.NODE_ENV === 'production-notugly') {
   )
 }
 
-app.use(express.static(path.resolve(__dirname, '../../', 'build')))
-app.get('/embed/:id', videoRoute)
-app.get('/play/:id', videoRoute)
+// app.use(express.static(path.resolve(__dirname, '../../', 'build')))
```
Comment (bent0b0x) [inline src/server/app.js]: Why is this removed? I think this would break a lot

File: src/server/app.js [hunk 9]
```diff
@@ -28,12 +28,15 @@ if (process.env.NODE_ENV === 'production-notugly') {
...
+app.get('/embed/:id', routeHelper)
+app.get('/', routeHelper)
 app.get('/oembed', oembedRoute)
-app.get('*', (req, res) => {
-  res.sendFile(path.resolve(__dirname, '../../', 'build', 'index.html'))
-})
+app.get('*', routeHelper)
+// app.get('*', (req, res) => {
+//   res.sendFile(path.resolve(__dirname, '../../', 'build', 'index.html'))
+// })
```
Comment (bent0b0x) [inline src/server/app.js]: I would think this comment would break loading the app when you visit `localhost:8080/upload` directly

**Comment by bent0b0x** [comment]
@geckoslair I think we need to use a formal templating engine instead of string concatenation. Additionally, I think there are changes to our route handler that need to be tested before merging this.

**Comment by bent0b0x** [comment]
https://github.com/ericf/express-handlebars

File: src/server/app.js (hunk 8 above)
Comment (geckoslair) [reply src/server/app.js]: i've removed because serves also index.html, do you know how to fix that?

File: src/server/app.js (hunk 9 above)
Comment (geckoslair) [reply src/server/app.js]: no, it works.

File: src/server/routes/index.js (hunk 7 above)
Comment (geckoslair) [reply src/server/routes/index.js]: sound better

File: src/server/routes/index.js (hunk 6 above)
Comment (geckoslair) [reply src/server/routes/index.js]: done

File: src/server/routes/index.js (hunk 5 above)
Comment (geckoslair) [reply src/server/routes/index.js]: ok

File: src/index.html (hunk 1 above)
Comment (geckoslair) [reply src/index.html]: ok

File: src/server/app.js (hunk 2 above)
Comment (geckoslair) [reply src/server/app.js]: ok

File: src/server/routes/index.js (hunk 3 above)
Comment (geckoslair) [reply src/server/routes/index.js]: ok

File: src/server/routes/index.js (hunk 4 above)
Comment (geckoslair) [reply src/server/routes/index.js]: ok

File: src/server/app.js (hunk 8 above)
Comment (bent0b0x) [reply src/server/app.js]: Without this I would not expect any of our asset loading to work (i.e. images will not load).

I'm sure there is an `express` api for excluding certain files. However, why do we need to exclude `/index.html`?

File: src/server/app.js (hunk 9 above)
Comment (bent0b0x) [reply src/server/app.js]: Ah, perhaps that is because `routeHelper` is doing the same thing?

File: src/server/app.js (hunk 8 above)
Comment (geckoslair) [reply src/server/app.js]: because it wouldn't get in the get request. I'll check with the handlebar features.
